In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('../Datasets/FinalFinal.csv')

In [3]:
df.head()


crop                     disease history   msp  climate_conditions  \
0   maize                         leaf blight  1750                  23   
1  cotton                         boll weevil  5150                  23   
2   wheat  leaf rust , spike blight , saw fly  1840                  24   
3   paddy         rice blast , shealth blast   1750                  25   
4  barley   bacterial blight , powdery mildew  1440                  35   

   rainfall  nitrogen  phosphorus  potassium  Season soil_conditions  Type  \
0       650       269         112        269  kharif       blacksoil     1   
1      1000       245          42         67  kharif       blacksoil     2   
2       750       202          73        173    rabi    alluvialsoil     3   
3       675        18           3         18  kharif    alluvialsoil     4   
4       550        60           7          7    rabi    alluvialsoil     5   

   Unnamed: 11  Unnamed: 12  
0          NaN          NaN  
1          NaN          NaN  
2          NaN          NaN  
3          NaN          NaN  
4          NaN          NaN

In [4]:
df.columns

Index(['crop', 'disease history', 'msp', 'climate_conditions', 'rainfall',
       'nitrogen', 'phosphorus', 'potassium', 'Season', 'soil_conditions',
       'Type', 'Unnamed: 11', 'Unnamed: 12'],
      dtype='object')

In [7]:
df = df[[ 'rainfall','nitrogen', 'phosphorus', 'potassium', 'Season', 'soil_conditions','crop']]

__Will ignore finalfinal.csv due to length of the dateset__

In [18]:
df = pd.read_csv('../Datasets/cpdata.csv')

In [19]:
df.head()


temperature   humidity        ph    rainfall label
0    20.879744  82.002744  6.502985  202.935536  rice
1    21.770462  80.319644  7.038096  226.655537  rice
2    23.004459  82.320763  7.840207  263.964248  rice
3    26.491096  80.158363  6.980401  242.864034  rice
4    20.130175  81.604873  7.628473  262.717340  rice

In [30]:
df.columns

Index(['temperature', 'humidity', 'ph', 'rainfall', 'label'], dtype='object')

In [20]:
df.corr()

temperature  humidity        ph  rainfall
temperature     1.000000 -0.076999  0.017024 -0.055143
humidity       -0.076999  1.000000 -0.002359  0.192074
ph              0.017024 -0.002359  1.000000 -0.288598
rainfall       -0.055143  0.192074 -0.288598  1.000000

In [26]:
df.label.unique()

array([28, 30, 11, 16, 23, 21,  9,  7,  4,  5,  6, 12, 14, 15, 17,  8, 10,
        3,  1,  0, 13,  2, 19, 20, 18, 22, 24, 25, 26, 27, 29])

In [24]:
df['label'] = df['label'].astype('category').cat.codes

In [25]:
df.corr()

temperature  humidity        ph  rainfall     label
temperature     1.000000 -0.076999  0.017024 -0.055143 -0.096094
humidity       -0.076999  1.000000 -0.002359  0.192074  0.285922
ph              0.017024 -0.002359  1.000000 -0.288598 -0.096932
rainfall       -0.055143  0.192074 -0.288598  1.000000 -0.003235
label          -0.096094  0.285922 -0.096932 -0.003235  1.000000

In [33]:
features = df[['temperature', 'humidity', 'ph', 'rainfall']]
labels = df['label']

In [34]:
from sklearn.model_selection import train_test_split

In [92]:
Xtrain, Xtest, Ytrain, Ytest = train_test_split(features,labels,test_size = 0.3,random_state =2)

In [90]:
from sklearn.tree import DecisionTreeClassifier
from sklearn import metrics

In [93]:
for i in range(5,15):
    dt = DecisionTreeClassifier(criterion="entropy",max_depth=i)
    dt.fit(Xtrain,Ytrain)
    predicted_values = dt.predict(Xtest)
    x =  metrics.accuracy_score(Ytest, predicted_values)
    print("DecisionTrees's Accuracy for depth ",i," is: ", x)



DecisionTrees's Accuracy for depth  5  is:  0.6924731182795699
DecisionTrees's Accuracy for depth  6  is:  0.7935483870967742
DecisionTrees's Accuracy for depth  7  is:  0.8709677419354839
DecisionTrees's Accuracy for depth  8  is:  0.9010752688172043
DecisionTrees's Accuracy for depth  9  is:  0.9193548387096774
DecisionTrees's Accuracy for depth  10  is:  0.9172043010752688
DecisionTrees's Accuracy for depth  11  is:  0.9043010752688172
DecisionTrees's Accuracy for depth  12  is:  0.9053763440860215
DecisionTrees's Accuracy for depth  13  is:  0.9043010752688172
DecisionTrees's Accuracy for depth  14  is:  0.9075268817204301


# Max accuracy of 91% for depth 9 for split 0.3 of dataset